### DISCUSSION TITLE
Error Code Meanings

### DISCUSSION THREAD

**Discussion by tetionet** [vanackej/risco-mqtt-local#59] [Q&A] [answered]
Hello,
I've recently cleaned my HA installation and I've discovered that Risco integration (fully local as I don't have anymore connection to the on pay cloud-service) is providing a couple of errors which I can't understand. These are errors from the Risco Panel (I've a prosys) and seems to be random (all entities are just monitored, no automation, no triggers based on that entitites, just monitoring them through lovelace).
``` 
2023-07-31 20:39:30.447 ERROR (MainThread) [homeassistant.components.risco] Error in Risco library: Risco error: N04
2023-08-01 06:46:03.459 ERROR (MainThread) [homeassistant.components.risco] Error in Risco library: Risco error: N04
2023-08-01 06:46:04.403 ERROR (MainThread) [homeassistant.components.risco] Error in Risco library: Timeout in command: CLOCK
```
In particular yesterday and today I didn't activate neither deactivated the alarm (so I'm almost sure these errors are not related to some specific action performed on local panel)
Is there any way to know what are the meaning of these errors?
Thank you

**Comment by vanackej** [comment]
Hi.
Are you sure you are using this addon ? I don't recognize the logs format.
I think you are using the official risco component, don't you ?

**Reply by tetionet** [reply]
You're right, I'm using the core risco integration with advanced configuration (for local use). I've contacted the core team which said that since these errors are simply output from local panel I should check the manual of the alarm system.
But as far as I know there is no manual relevant to these code messages, so I was wondering if you have the meaning of these codes (you've already replied some time ago about another "Nxx" error

**Reply by vanackej** [reply, marked as answer]
You can find the error code constants and their meaning here : https://github.com/vanackej/risco-lan-bridge/blob/main/src/constants.ts
N04 is a CRC error, your pin or panel id must be incorrect
I close this issue as it is not related to this project
